app/shop/pages/manageProducts/index.tsx

```diff
@@ -44,14 +45,24 @@ const ShopProducts = () => {
   const [{ products }] = useQuery(getProducts, {
     where: { shopId: user!.shop!.id },
   })
+  const [productList, setProductList] = useState(products)
+  const [deleteProductMutation] = useMutation(deleteProduct)
+
   const [value, setValue] = useState('all')
   const { available, sold } = useMemo(() => {
-    const sold = products.filter((product) => isProductSoldOut(product))
-    const available = products.filter((product) => !isProductSoldOut(product))
+    const sold = productList.filter((product) => isProductSoldOut(product))
+    const available = productList.filter(
+      (product) => !isProductSoldOut(product)
+    )
     return { available, sold }
-  }, [products])
+  }, [productList])
   const visibleProducts =
-    value === 'all' ? products : value === 'sold out' ? sold : available
+    value === 'all' ? productList : value === 'sold out' ? sold : available
+
+  const handleProductDelete = async (id: number) => {
+    await deleteProductMutation({ id })
+    setProductList(productList.filter((product) => product.id != id))
```

Comment:
Remove the `productList` state and use [this](https://github.com/2110336-SoftwareEngineering2-2021-2/sec33_jim_carry/blob/f5c530f9e614debd70053d117de5e9190fd7a634/app/users/components/AddressList.tsx#L30) instead.